atomic_reactor/inner.py

```diff
@@ -626,12 +626,16 @@ def parent_images_to_str(self):
     def image(self):
         return self.user_params['image_tag']
 
-    @property
-    def build_process_failed(self):
+    @functools.cached_property
+    def osbs(self) -> OSBS:
+        return get_openshift_session(self.conf, self.namespace)
+
+    @functools.cached_property
+    def build_process_failed(self) -> bool:
         """
         Has any aspect of the build process failed?
         """
-        return self.data.plugin_failed
+        return self.osbs.build_has_any_failed_tasks(self.pipeline_run_name)
```

Comment:
And another question, mainly for @tkdchen:

Is it OK that this will return False when the build is cancelled? It kinda looks like the cancel_reservation plugin would expect this to be True in that case

https://github.com/containerbuildsystem/atomic-reactor/blob/e1cb2b82527e5097009c4705474a5bc963f5826c/atomic_reactor/plugins/cancel_build_reservation.py#L52-L59

Reply:
Why is False returned when build is cancelled?

Reply:
IMO 'failed' and 'cancelled' are different outcomes. A cancelled build is not a failed build. Should this return True for both failed and cancelled builds?

Reply:
we want to always call CGRefundBuild with correct state

failed build should call it with 'FAILED', canceled with 'CANCELED'

Reply:
Updated with suggestion about returning (failed, cancelled) from the same method. Cancelled counts as failed (the method can't return (False, True)